### PR TITLE
Add blocks conflicting with ChainLocks to block index

### PR DESCRIPTION
**Pull request by codablock**

### CODE REVIEW

**Comment by codablock** [comment]
After short discussion in Slack, I decided to revert this and only include the change to the LogPrintf so that it prints the conflicting hash.

**Comment by codablock** [comment]
After more discussion, I applied suggested code from @UdjinM6 which allows us to add the block to the block index but without going through the full logic of `InvalidateBlock` (which causes undesired signals)

File: src/validation.cpp [hunk 1]
```diff
@@ -3100,9 +3100,13 @@ CBlockIndex* AddToBlockIndex(const CBlockHeader& block)
...
     pindexNew->nTimeMax = (pindexNew->pprev ? std::max(pindexNew->pprev->nTimeMax, pindexNew->nTime) : pindexNew->nTime);
     pindexNew->nChainWork = (pindexNew->pprev ? pindexNew->pprev->nChainWork : 0) + GetBlockProof(*pindexNew);
-    pindexNew->RaiseValidity(BLOCK_VALID_TREE);
-    if (pindexBestHeader == NULL || pindexBestHeader->nChainWork < pindexNew->nChainWork)
-        pindexBestHeader = pindexNew;
+    if (fValid) {
+        pindexNew->RaiseValidity(BLOCK_VALID_TREE);
+        if (pindexBestHeader == NULL || pindexBestHeader->nChainWork < pindexNew->nChainWork)
+            pindexBestHeader = pindexNew;
+        } else {
```
Comment (UdjinM6) [inline src/validation.cpp]:
```suggestion
    } else {
```

**Comment by UdjinM6** [comment]
Pls see https://github.com/UdjinM6/dash/commit/67857a1136cb5f64bd95c798ac604e21b0de7592 and https://github.com/UdjinM6/dash/commit/dc3e71b312e8640b13a57c43772d3b4c4aae740b

**Comment by codablock** [comment]
Replaced the commits with https://github.com/UdjinM6/dash/commit/67857a1136cb5f64bd95c798ac604e21b0de7592 and https://github.com/UdjinM6/dash/commit/dc3e71b312e8640b13a57c43772d3b4c4aae740b :)

**Pull request merged**